src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +240,209 @@ def truncate_table(self, table):
+            return True
+        return False
+
+    def create_table_using_native_schema_autodetection(
+        self,
+        table: Table,
+        file: File,
+    ) -> None:
+        """
+        Create a SQL table, automatically inferring the schema using the given file via native database support.
+        :param table: The table to be created.
+        :param file: File used to infer the new table columns.
+        """
+        supported_config = self.NATIVE_AUTODETECT_SCHEMA_CONFIG.get(file.location.location_type)
+        if not supported_config:
+            raise ValueError("Unsupported file location.")
+        return supported_config["method"](table=table, file=file)  # type: ignore
+
+    # Require skipcq because method overriding we need param target_table
+    def is_native_path_available(  # skipcq: PYL-R0201
+        self,
+        source_dataset: File | Table,  # skipcq PYL-W0613
+    ) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        if isinstance(source_dataset, Table):
+            return False
+        file_type = NATIVE_PATHS_SUPPORTED_FILE_TYPES.get(source_dataset.type.name)
+        location_type = self.NATIVE_PATHS.get(source_dataset.location.location_type)
+        return bool(location_type and file_type)
+
+    def load_file_to_table_natively(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        :param if_exists: Overwrite file if exists. Default False
+        """
+        method_name = self.NATIVE_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+        else:
+            raise DatabaseCustomError(
+                f"No transfer performed since there is no optimised path "
+                f"for {source_file.location.location_type} to bigquery."
+            )
+
+    def load_gs_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Transfer data from google cloud store to bigquery
+
```

Comment:
nitpick: It might be better if we can unify whether we want to add a blank line before `:param`. I would suggest adding one blank line like this one.

Reply:
https://github.com/astronomer/apache-airflow-provider-transfers/pull/66/commits/e83030f053afc687621114a710444fdaf51b3a0c